capa/ida/plugin/form.py

```diff
@@ -608,14 +647,40 @@ def on_load_rule(rule_path, i, total):
 
         return True
 
-    def load_capa_results(self, use_cache=False):
+    def load_capa_results(self, use_volatile_cache, use_persistent_cache):
         """run capa analysis and render results in UI
 
         note: this function must always return, exception or not, in order for plugin to safely close the IDA
         wait box
         """
-        if not use_cache:
+        if use_persistent_cache:
+            try:
+                update_wait_box("loading rules")
+                if not self.load_capa_rules():
+                    return False
+                assert self.ruleset_cache is not None
+
+                if compute_ruleset_cache_identifier(self.ruleset_cache) != capa.ida.helpers.load_rules_cache_id():
+                    logger.error(
+                        "The ruleset cache and the local rule definitions differ. Please reanalyze the program."
```

Comment:
- My first thought is **not** to keep the cached rule set and cached results in sync
    - Suggested workflow:
      - Check for cached results
        - If found, load cached results ignoring cached rule set
          - Update bottom left status message to say "Loaded existing capa results from..."
          - Note: this should also save use CPU cycles by not loading the cached rule set when we don't use it
        - Else, load cached rule set, run anlaysis, cache results
          - Update bottom left status message to say "capa rules: <rule_path> (<rule_count> rules)"

Use-case: user creates a new rule using the rule generator and adds it to their rule path. User is now forced to re-analyze even if they do not want to spend time waiting for the new rule to be included.

Open to discussion here as I see benefits to both approaches.

Reply:
Changed this to warn the user, but ultimately letting them decide to leave or reanalyze. What do you think of this?

Reply:
![2023-02-03_11-06-03_ida](https://user-images.githubusercontent.com/17606537/216572168-ad2aa780-17e3-4e44-a301-59e6d151043f.png)